pkg/server/plugin/datastore/sql/migration.go

```diff
@@ -86,7 +85,7 @@ func migrateDB(db *gorm.DB, dbType string, disableMigration bool, log hclog.Logg
 	}
 
 	if disableMigration {
-		if err = isDisabledMigrationAllowed(dbCodeVersion); err != nil {
+		if err = isDisabledMigrationAllowed(codeVersion, dbCodeVersion); err != nil {
```

Comment:
maybe this is a clue that `codeVersion` should no longer be a package level variable and should be locally scoped to migrateDB.... then all the akwardness around the naming can go away (e.g. `thisCodeVersion`).

Reply:
💯 

Yea, this is one of many smells in the compatibility checks that we have in this package.

How about a compromise - I'll stage changes for cleaning all this mess up, and we can merge them immediately following 0.12.0 release?

Reply:
Sounds good to me!